### PR TITLE
feat: add Instafeed.js types

### PR DESCRIPTION
**Pull request by IncognitaDev**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/instafeed.js/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (jakebailey) [inline types/instafeed.js/tsconfig.json:15]: This setting should not be set.

Comment (IncognitaDev) [reply types/instafeed.js/tsconfig.json]: Fixed!

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/instafeed.js/instafeed.js-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 365b4ec54535f683c3440d293d2f7e8e5dcd01aa
</p>

File: types/instafeed.js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    _basicRender(data: T): string;
+    _valueForKeyPath(keyPath: string, data: unknown): string;
+    _fail(err: unknown): void;
+    _start(): void;
+    _finish(): void;
+    _success(data: Instafeed.InstagramResponseData): void;
+    _makeApiRequest(url: string, callback: (err: unknown, data: Instafeed.InstagramResponseData) => void): void;
+    _getAccessToken(callback: (err: unknown, token: string) => void): void;
+    _debug(): void;
+    _runHook(hookname: string, data: unknown): boolean;
```
[the remark this answers is not in the document]
Comment (IncognitaDev) [reply types/instafeed.js/index.d.ts:28]: I dont know if i should include this, but i search and find a tons of types that have methods with this private prefix or even private keyworld, but I can remove, looks better to me.

[the remark this answers is not in the document]
Comment (IncognitaDev) [reply types/instafeed.js/index.d.ts:28]: I think `_state` and `_options` can be usefull in some case, can i left they?

[the remark this answers is not in the document]
Comment (peterblazejewicz) [reply types/instafeed.js/index.d.ts]: After looking into code, I'd strip everything except of constructor. Usage is as simple as creating instance passing optional options.
You're welcome to leave whatever you want, but when doing so, add documentation (JSDoc) what is the purpose (as this is missing from readme and docs).

[the remark this answers is not in the document]
Comment (IncognitaDev) [reply types/instafeed.js/index.d.ts]: I chose to remove

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/instafeed.js/instafeed.js-tests.ts
1. types/instafeed.js/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 76934bccfaf914f373b597f8ff0b22823a0adabd
</p>

File: types/instafeed.js/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "projects": [
+        "https://instafeedjs.com"
+    ],
+    "devDependencies": {
+        "@types/instafeed.js": "workspace:."
+    },
+    "owners": [
+        {
+            "name": "Luis Sousa",
+            "githubUsername": "DefinitelyTyped"
```
Comment (peterblazejewicz) [inline types/instafeed.js/package.json:14]: shouldn't this read:
```suggestion
            "githubUsername": "IncognitaDev"
```
(only user accounts can be owners, DT just takes ownerships if types definition goes derelict)

Comment (IncognitaDev) [reply types/instafeed.js/package.json]: I think this has been added by the create command

Comment (jakebailey) [reply types/instafeed.js/package.json]: Yeah, typically this means that your local git information was not enough to determine the right value (so it picks a default). The default should probably be the empty array (no owners used to be illegal).

**Review by jakebailey — APPROVED** [review]

**Comment by IncognitaDev** [comment]
Ready to merge

**Review by peterblazejewicz — APPROVED** [review]

**Pull request merged**